core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -223,21 +225,22 @@ private Object visitDataReadCommand(InvocationContext ctx, AbstractDataCommand c
       final Object key = command.getKey();
       CompletionStage<Void> stage = entryFactory.wrapEntryForReading(ctx, key, command.getSegment(),
             ignoreOwnership(command) || canRead(command), command.hasAnyFlag(FlagBitSets.ALREADY_HAS_LOCK)
-                  || (isPessimistic && command.hasAnyFlag(FlagBitSets.FORCE_WRITE_LOCK)));
+                  || (isPessimistic && command.hasAnyFlag(FlagBitSets.FORCE_WRITE_LOCK)), CompletableFutures.completedNull());
       return makeStage(asyncInvokeNext(ctx, command, stage)).thenApply(ctx, command, dataReadReturnHandler);
    }
 
    @Override
    public Object visitGetAllCommand(InvocationContext ctx, GetAllCommand command) throws Throwable {
       boolean ignoreOwnership = ignoreOwnership(command);
-      AggregateCompletionStage<Void> aggregateCompletionStage = null;
+      CompletableFuture<Void> initialStage = new CompletableFuture<>();
+      CompletionStage<Void> currentStage = initialStage;
       for (Object key : command.getKeys()) {
-         CompletionStage<Void> stage = entryFactory.wrapEntryForReading(ctx, key, keyPartitioner.getSegment(key),
-               ignoreOwnership || canReadKey(key), false);
-         aggregateCompletionStage = accumulateStage(stage, aggregateCompletionStage);
+         currentStage = entryFactory.wrapEntryForReading(ctx, key, keyPartitioner.getSegment(key),
+                                                         ignoreOwnership || canReadKey(key), false,
+                                                         CompletableFutures.completedNull());
```

Comment:
`currentStage`?

Reply:
Good catch, I thought I had fixed this one.

Reply:
Turns out I had fixed it only in the 9.4.x branch...

But before I noticed that, seeing 2 bugs in the same method convinced me add a `getAll` test, and that revealed yet another issue: `wrapEntryForReading` sometimes returning `completedNull()` instead of `previousStage`.

Thanks Pedro ;)